pkg/retrieval/retrieval_test.go

```diff
@@ -297,19 +299,19 @@ func TestRetrievePreemptiveRetry(t *testing.T) {
 			streamtest.WithMiddlewares(
 				func(h p2p.HandlerFunc) p2p.HandlerFunc {
 					return func(ctx context.Context, peer p2p.Peer, stream p2p.Stream) error {
+						failM.Lock()
+						defer failM.Unlock()
```

Comment:
Would it be better to explicitly unlock here before the server handler is called?